src/backend/InvenTree/templates/js/translated/label.js

```diff
@@ -56,145 +58,103 @@ function printLabels(options) {
-                    '{% trans "No label templates found which match the selected items" %}',
-                );
-                return;
+    function getPrintingFields(plugin_id, callback) {
+        let url = '{% url "api-label-print" %}' + `?plugin=${plugin_id}`;
+
+        inventreeGet(
+            url,
+            {
+                plugin: plugin_id,
+            },
+            {
+                method: 'OPTIONS',
+                success: function(response) {
+                    let fields = response?.actions?.POST ?? {};
+                    callback(fields);
+                }
             }
+        );
+    }
 
-            labelTemplates = response;
-        }
-    });
+    // Callback when a particular label printing plugin is selected
+    function onPluginSelected(value, name, field, formOptions) {
 
-    // Request a list of available label printing plugins from the server
-    let plugins = [];
-    inventreeGet(`/api/plugins/`, { mixin: 'labels' }, {
-        async: false,
-        success: function (response) {
-            plugins = response;
+        if (value == pluginId) {
+            return;
         }
-    });
 
-    let header_html = "";
+        pluginId = value;
 
-    // show how much items are selected if there is more than one item selected
-    if (options.items.length > 1) {
-        header_html += `
-            <div class='alert alert-block alert-info'>
-            ${options.items.length} ${options.plural_name} {% trans "selected" %}
-            </div>
-        `;
-    }
+        // Request new printing options for the selected plugin
+        getPrintingFields(value, function(fields) {
+            formOptions.fields = getFormFields(fields);
+            updateForm(formOptions);
 
-    const updateFormUrl = (formOptions) => {
-        const plugin = getFormFieldValue("_plugin", formOptions.fields._plugin, formOptions);
-        const labelTemplate = getFormFieldValue("_label_template", formOptions.fields._label_template, formOptions);
-        const params = $.param({ plugin, [options.key]: options.items })
-        formOptions.url = `${options.url}${labelTemplate ?? "1"}/print/?${params}`;
+            // workaround to fix a bug where one cannot scroll after changing the plugin
+            // without opening and closing the select box again manually
+            $("#id__plugin").select2("open");
+            $("#id__plugin").select2("close");
+        });
     }
 
-    const updatePrintingOptions = (formOptions) => {
-        let printingOptionsRes = null;
-        $.ajax({
-            url: formOptions.url,
-            type: "OPTIONS",
-            contentType: "application/json",
-            dataType: "json",
-            accepts: { json: "application/json" },
-            async: false,
-            success: (res) => { printingOptionsRes = res },
-            error: (xhr) => showApiError(xhr, formOptions.url)
-        });
+    const baseFields = {
+        template: {},
+        plugin: {},
```

Comment:
Is that right, that plugin is now expected to be a numeric id and not the plugin slug anymore? Is there still an option then to print with only the slug, or do I need two requests then? Because the slug is guessable by 3rd party extensions, the id not, that can be different between instances.

Reply:
The plugin needs to be specified by *id* now - this is so that we can use a foreignkeyrelatedfield on the serializer - and reference to the *pluginconfig* object - and expose the serialized fields to the documented API.

I think that the downside (have to use ID values rather than "slug" values) is acceptable because it simplifies the implementation and increases consistency...

Reply:
Ok, so printing with the machine plugin or the pdf printer plugin from a third party tool, now requires a second request to first ask for the corresponding id if that plugin?

How have you handled this now in the template editor to preview the file? Is there now a shared endpoint with the reports to preview a label directly rather than needing to request a pdf from a printing plugin first and then making a second request to grab the file?

Reply:
Ok, so the plugin should be selected by slug, not by ID?